CONTRIBUTING.md

```diff
@@ -40,17 +40,40 @@ Then Gradle task rule will kick in and do the following:
 This works both for tasks named `test` and `latestDepTest`.
 But currently does not work for other custom test tasks, such as those created by test sets plugin.
 
-### Code Style
+### Style guideline
 
-This project includes a `.editorconfig` file for basic editor settings.  This file is supported by most common text editors.
+We follow the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html). 
+Our build will fail if source code is not formatted according to that style.
 
-Java files must be formatted using [google-java-format](https://github.com/google/google-java-format).  Please run the following task to ensure files are formatted before committing:
+To verify code style manually run the following command, 
+which uses [google-java-format](https://github.com/google/google-java-format) library:
 
-```shell 
-./gradlew googleJavaFormat
-```
+`./gradlew verifyGoogleJavaFormat`
 
-Other source files (Groovy, Scala, etc) should ideally be formatted by Intellij Idea's default formatting, but are not enforced.
+or on Windows
+
+`gradlew.bat verifyGoogleJavaFormat`
+
+Instead of fixing style inconsistencies by hand, you can run gradle task `googleJavaFormat`
+to automatically fix all found issues:
+
+`./gradlew googleJavaFormat`
+
+or on Windows
+
+`gradlew.bat googleJavaFormat`
+
+#### Pre-commit hook
+To completely delegate code style formatting to the machine, 
+you can add [git pre-commit hook](https://git-scm.com/docs/githooks).
+We provide an example script in `buildscripts/pre-commit` file.
+Just copy or symlink it into `.git/hooks` folder.
```

Comment:
include suggested command?

Reply:
Include an example of `cp` command? I expect any developer who is wishing to contribute to this project knows how to do that.

Reply:
I was thinking the symlink, but good point...